internal/subscriber/core_test.go

```diff
@@ -0,0 +1,260 @@
+				subscriberID: subID,
+			},
+			msgs: []string{"a", "b"},
+		},
+		{
+			name: "Create new subscriber with getConsumer error Failure",
+			fields: fields{
+				core: getCore(store, core, offsetCore, ch),
+			},
+			args: args{
+				ctx:          ctx,
+				subscriberID: subID,
+				subscription: &subscrption,
+				requestCh:    requestCh,
+				ackCh:        ackCh,
+				modAckCh:     modAckCh,
+			},
+			expected: nil,
+			wantErr:  true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			store.EXPECT().GetConsumer(gomock.Any(), gomock.Any()).
+				DoAndReturn(func(arg0 context.Context, arg1 messagebroker.ConsumerClientOptions) (messagebroker.Consumer, error) {
+					if tt.wantErr {
+						return nil, errors.New("Test Error")
+					}
+					return cs, nil
+				}).AnyTimes()
+			count := 0
+			cs.EXPECT().ReceiveMessages(gomock.Any(), messagebroker.GetMessagesFromTopicRequest{
+				NumOfMessages: 10,
+				TimeoutMs:     100,
+			}).DoAndReturn(
+				func(arg0 context.Context, arg1 messagebroker.GetMessagesFromTopicRequest) (*messagebroker.GetMessagesFromTopicResponse, error) {
+					if count == 0 {
+						count++
+						return &messagebroker.GetMessagesFromTopicResponse{Messages: getMockReceivedMessages(tt.msgs)}, nil
+					}
+					return &messagebroker.GetMessagesFromTopicResponse{}, nil
+				}).AnyTimes()
+			sub, err := tt.fields.core.NewSubscriber(tt.args.ctx, tt.args.subscriberID, tt.args.subscription, timeoutInMs,
+				maxOutstandingMessages, maxOutstandingBytes, tt.args.requestCh, tt.args.ackCh, tt.args.modAckCh)
+			assert.Equal(t, err != nil, tt.wantErr)
+			assert.True(t, EqualOnly(sub, tt.expected, []string{"subscription", "subscriberID"}))
+			if err == nil {
+				sub.Stop()
+			}
+		})
+	}
+}
+
+func TestNewCore(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockBrokerStoreCore := mocks.NewMockIBrokerStore(ctrl)
+	mockSubscriptionCore := sCore.NewMockICore(ctrl)
+	mockOffsetCore := oCore.NewMockICore(ctrl)
+	mockCache := mCache.NewMockICache(ctrl)
+
+	type args struct {
+		bs               brokerstore.IBrokerStore
+		subscriptionCore subscription.ICore
+		offsetCore       offset.ICore
+		ch               cache.ICache
+	}
+	tests := []struct {
+		name string
+		args args
+		want ICore
+	}{
+		{
+			name: "Get New subscriber Core",
+			args: args{
+				bs:               mockBrokerStoreCore,
+				subscriptionCore: mockSubscriptionCore,
+				offsetCore:       mockOffsetCore,
+				ch:               mockCache,
+			},
+			want: &Core{
+				bs:               mockBrokerStoreCore,
+				subscriptionCore: mockSubscriptionCore,
+				offsetCore:       mockOffsetCore,
+				ch:               mockCache,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := NewCore(tt.args.bs, tt.args.subscriptionCore, tt.args.offsetCore, tt.args.ch)
+			assert.Equalf(t, tt.want, got, "NewCore(%v, %v, %v, %v)", tt.args.bs, tt.args.subscriptionCore, tt.args.offsetCore, tt.args.ch)
```

Comment:
why are we using Equalf and not Equal..?

Reply:
I though EqualF will print detailed difference if Equalf assertion failed. since all the variable are mocked it is not required here. removing Equalf